main.cpp

```diff
@@ -0,0 +1,127 @@
+      " your password: ", "Please enter a timezone offset: ", "Enter a file path for the config file: "};
+
+    cout << "\nCONFIG FILE CREATION\n\n";
+    string results[SIZE];
+
+    // prompt for all questions that require input
+    for(int i = 0; i < SIZE; i++) {
+        cout << "\t" + PROMPTS[i];
+        getline(cin, results[i]);
+        while(results[i].length() == 0) {
+            if(i == (SIZE - 1)){
+                results[i] = FILE_PATH;
+                cout << "\t( File path set to default location )";
+            } else {
+                cout << "You did not enter any input. Please try again.\n";
+                cout << PROMPTS[i];
+                cin >> results[i];
+            }
+        }
+    }
+
+    // create/open file and print headers and user input
+    ofstream config(FILE_PATH);
+    if(config) {
+        for(int i = 0; i < SIZE; i++){
+            config << HEADERS[i] + "\n\t" + results[i] + "\n";
+        }
+    }else{
+        cout << "Error in file creation.";
+    }
+
+    // make sure to close the file!
+    config.close();
+    cout << "\n\tFile Created!";
+}
+void edit(string field) {
+    const int SIZE = 5;
+
+    cout << "\nCONFIG FILE EDIT\n\n";
+    cout << "\tEnter a filepath, or press 'Enter' to open the default path: ";
+    string path;
+    getline(cin, path);
+    if(path.length() == 0) {
+        path = FILE_PATH;
+    }
+
+    // open file and store data for editing
+    ifstream config;
+    config.open(path);
+    if(config){
+        string text; string headers[SIZE]; string values[SIZE]; int i = 0;
+        while(config >> text){
+            getline(config, headers[i]);
+            getline(config, values[i]);
+            i++;
+        }
+        // file closes after it is finished reading input
+        config.close();
+
+        // prompt user to change given field
+        cout << "\tEnter a new value for the " + field + " field: ";
+        if(field == "name"){
+            cin >> values[0];
+            values[0] = "\t" + values[0];
+        } else if(field == "email"){
+            cin >> values[1];
+            values[1] = "\t" + values[1];
+        } else if(field == "password"){
+            cin >> values[2];
+            values[2] = "\t" + values[2];
+        } else if(field == "timezone"){
+            cin >> values[3];
+            values[3] = "\t" + values[3];
+        } else {
+            cin >> values[4];
+            values[4] = "\t" + values[4];
+        }
+        ofstream config(path);
+        for(i = 0; i < SIZE; i++){
+            config << headers[i] << "\n\t" << values[i] << "\n";
+        }
+        cout << "\n\tFile Updated!";
+        config.close();
+    } else {
+        cout << "Error in locating file.";
+    }
+}
+
+int main(int argc, char *argv[]) {
+    // initialize variables with given arguments
+    string arg1 =  static_cast<string>(argv[1]);
+    string arg2 = "";
+    if(argv[2] != nullptr){
+        arg2 = static_cast<string>(argv[2]);
+    }
```

Comment:
This is fine you can also use the count of argc to help this. If it is edit there are 2 elements in argv
1. execution path
2. first argument

if argc == 3 then
1. execution path
2. first argument (init, edit)
3. second argument (name, email, ...)